src/v/redpanda/admin_server.cc

```diff
@@ -2605,6 +2606,111 @@ admin_server::mark_transaction_expired_handler(
       });
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::get_reconfigurations_handler(std::unique_ptr<ss::http::request>) {
+    using reconfiguration = ss::httpd::partition_json::reconfiguration;
+
+    auto& in_progress
+      = _controller->get_topics_state().local().updates_in_progress();
+
+    std::vector<model::ntp> ntps;
+    ntps.reserve(in_progress.size());
+
+    for (auto& [ntp, status] : in_progress) {
+        ntps.push_back(ntp);
+    }
+
+    auto const deadline = model::timeout_clock::now() + 5s;
+
+    auto [reconfiguration_states, reconciliations]
+      = co_await ss::when_all_succeed(
+        _controller->get_api().local().get_partitions_reconfiguration_state(
+          ntps, deadline),
+        _controller->get_api().local().get_global_reconciliation_state(
+          ntps, deadline));
+
+    if (reconfiguration_states.has_error()) {
+        vlog(
+          logger.info,
+          "unable to get reconfiguration status: {}({})",
+          reconfiguration_states.error().message(),
+          reconfiguration_states.error());
+        throw ss::httpd::base_exception(
+          fmt::format(
+            "unable to get reconfiguration status: {}({})",
+            reconfiguration_states.error().message(),
+            reconfiguration_states.error()),
+          ss::http::reply::status_type::service_unavailable);
+    }
+    std::vector<reconfiguration> all_reconfigurations;
+    for (auto& s : reconfiguration_states.value()) {
+        reconfiguration r;
+        r.ns = s.ntp.ns;
+        r.topic = s.ntp.tp.topic;
+        r.partition = s.ntp.tp.partition;
+
+        for (const model::broker_shard& bs : s.current_assignment) {
+            ss::httpd::partition_json::assignment assignment;
+            assignment.core = bs.shard;
+            assignment.node_id = bs.node_id;
+            r.current_replicas.push(assignment);
+        }
+
+        for (const model::broker_shard& bs : s.previous_assignment) {
+            ss::httpd::partition_json::assignment assignment;
+            assignment.core = bs.shard;
+            assignment.node_id = bs.node_id;
+            r.previous_replicas.push(assignment);
+        }
+
+        size_t left_to_move = 0;
+        size_t already_moved = 0;
```

Comment:
not blocking this pr, but it would be really great to have these available as metrics and drilled down per node to be able to see a dynamic.

Reply:
good idea, we will do it as a follow up